### PR TITLE
🎉 All java connectors: Added configValidator to check, discover, read and write calls

### LINKED ISSUE #4352: Validate that input config matches spec inside all java connectors by default
## Tell us about the problem you're trying to solve
Today if the input config in a connector does not match the connector's `spec.json` then the connector will not validate it. 

## Describe the solution you’d like
Modify the `Destination` and `Source` java abstractions to automatically validate that the input config to `discover`, `read`, `check`, and `write` matches the connector's spec. This way the child class does not need to do anything about validation -- it is automatically handled for it. 

## Describe the alternative you’ve considered or used
Test for this in acceptance tests, which is complementary to this issue. 

## Additional context

### LINKED ISSUE #4748: Follow up ticket: bump java connectors versions and release it
Java base code part had been updated as part of the https://github.com/airbytehq/airbyte/pull/4699
Now need to bump up versions and release all of java connectors.

P.S. Integration tests for some of the connectors (mostly destination) had been failed even bore adding this check.
So as part of this task, you will need to to check why test fails and fix them. For more details of failed test please check the latest comments from the linked ticket (https://github.com/airbytehq/airbyte/pull/4699)

### PR DESCRIPTION
**Pull request by etsybaev**
## What
Today if the input config in a connector does not match the connector's spec.json then the connector will not validate it.

## How
Modified IntegrationRunner to verify it while check, discover, read and write calls

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Secrets are annotated with `airbyte_secret` in the connector's spec
- [ ] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [x] Code reviews completed
- [x] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [x] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] No major blockers
- [x] PR merged into master branch
- [ ] Follow up tickets have been created
- [x] Associated tickets have been closed & stakeholders notified
</p>
</details>

<details><summary> <strong> Connector Generator checklist </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Comment by cgardens** [comment]
@etsybaev this is cool. you may be able to make your life a little bit easier if you leverage the `JsonSchemaValidator` in the `airbyte-json-validation` module. That validator can handle all of the jsonschema validation details so that you can just focus translating them into something human readable. Just a thought.

**Comment by etsybaev** [comment]
> @etsybaev this is cool. you may be able to make your life a little bit easier if you leverage the `JsonSchemaValidator` in the `airbyte-json-validation` module. That validator can handle all of the jsonschema validation details so that you can just focus translating them into something human readable. Just a thought.

coll, many thanks for the hint. I didn't see it. Will take a look

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java [hunk 1]
```diff
@@ -54,6 +56,8 @@
   private final Integration integration;
   private final Destination destination;
   private final Source source;
+  // private ConfigAgainstSpecValidator configAgainstSpecValidator;
```
Comment (yaroslav-hrytsaienko) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:59]: please remove unneeded comment

Comment (etsybaev) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java]: done, thanks

**Comment by etsybaev** [comment]
@sherifnada will we need to bump versions and re-publish all java connectors as part of this ticket?

**Comment by sherifnada** [comment]
@subodh1810 Can you take a look ?

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java [hunk 2]
```diff
@@ -68,13 +71,26 @@ public IntegrationRunner(Source source) {
...
+    validator = new JsonSchemaValidator();
+  }
+
+  @VisibleForTesting
+  IntegrationRunner(IntegrationCliParser cliParser,
+                    Consumer<AirbyteMessage> outputRecordCollector,
+                    Destination destination,
+                    Source source,
+                    JsonSchemaValidator jsonSchemaValidator) {
+
```
Comment (tuliren) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:91]: Nit. This extra line seems unnecessary.

**Comment by subodh1810** [comment]
@sherifnada yup I can

**Review by tuliren — APPROVED** [review]
Looks great! Thank you~~

A general nitpick. There are quite a few code changes from line wrapping, which seem unnecessary. We usually allow up to 150 characters per line ([code style](https://github.com/airbytehq/airbyte/blob/master/docs/contributing-to-airbyte/code-style.md)).

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java (hunk 2 above)
Comment (etsybaev) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:91]: Ok, I will update it. I just have the plugin enabled that automatically formats code by "google style" for java code. So assume that long lines were splitting by this plugin.
Thanks

**Review by subodh1810 — APPROVED** [review]
Looks fine, just that I dont think this comment https://github.com/airbytehq/airbyte/pull/4699#issuecomment-879365736 triggered all the tests and if you want to run the integration tests then a single comment for each connector would work

**Comment by etsybaev** [comment]
Some tests for connectors fail (most of them seems to fails with similar errors), but none of them are because of added changes, so merging it

**Comment by cgardens** [comment]
Nice! Awesome when a 3 file change can have this much impact.

File: airbyte-integrations/bases/base-java/build.gradle [hunk 3]
```diff
@@ -7,6 +7,8 @@ dependencies {
     implementation 'commons-cli:commons-cli:1.4'
 
     implementation project(':airbyte-protocol:models')
+    implementation project(':airbyte-queue')
```
Comment (michel-tricot) [inline airbyte-integrations/bases/base-java/build.gradle:10]: Why did we add the deps to `airbyte-queue`?

Comment (etsybaev) [reply airbyte-integrations/bases/base-java/build.gradle:10]: Hmmm...as part of this PR I explicitly added the only implementation project(":airbyte-json-validation").
Have no idea where "airbyte-queue" comes from. Maybe as part of auto-merging masker branch... some kind of merge conflicts but I didn't see it. Shall we remove it?

Comment (ChristopheDuong) [reply airbyte-integrations/bases/base-java/build.gradle:10]: I'm removing this here: https://github.com/airbytehq/airbyte/pull/4881/files#r673985881

**Pull request merged**